pkg/guacanalytics/patchPlanning.go

```diff
@@ -158,35 +180,52 @@ func exploreIsDependencyFromDepPkg(ctx context.Context, gqlclient graphql.Client
 		return nil
 	}
 
-	dfsNVersion, seenVersion := q.nodeMap[isDependency.Package.Namespaces[0].Names[0].Versions[0].Id]
-	dfsNName, seenName := q.nodeMap[isDependency.Package.Namespaces[0].Names[0].Id]
+	addNodeToQueue(q, q.now, q.nowNode.Depth, PackageVersion, false, nil, isDependency.Package.Namespaces[0].Names[0].Versions[0].Id)
+	addNodeToQueue(q, q.now, q.nowNode.Depth, PackageName, false, []string{isDependency.Package.Namespaces[0].Names[0].Versions[0].Version}, isDependency.Package.Namespaces[0].Names[0].Id)
 
-	if !seenVersion {
-		dfsNVersion = DfsNode{
-			Parent:   q.now,
-			Depth:    q.nowNode.Depth + 1,
-			NodeType: "packageVersion",
-		}
-		q.nodeMap[isDependency.Package.Namespaces[0].Names[0].Versions[0].Id] = dfsNVersion
+	return nil
+}
+
+func exploreIsOccurrenceFromSubject(ctx context.Context, gqlclient graphql.Client, q *queueValues, isOccurrence model.NeighborsNeighborsIsOccurrence) {
+	addNodeToQueue(q, q.now, q.nowNode.Depth, Artifact, false, nil, isOccurrence.Artifact.Id)
+}
+
+func exploreHasSLSAFromArtifact(ctx context.Context, gqlclient graphql.Client, q *queueValues, hasSLSA model.NeighborsNeighborsHasSLSA) {
+	// Check that the subject is not the node inputted itself and being re-added to the queue unnecessarily
+	if q.now != hasSLSA.Subject.Id {
+		addNodeToQueue(q, q.now, q.nowNode.Depth, Artifact, true, nil, hasSLSA.Subject.Id)
 	}
+}
 
-	if !seenName {
-		dfsNName = DfsNode{
-			Parent:       q.now,
-			Depth:        q.nowNode.Depth + 1,
-			NodeType:     "packageName",
-			nodeVersions: []string{isDependency.Package.Namespaces[0].Names[0].Versions[0].Version},
+func exploreIsOccurrenceFromArtifact(ctx context.Context, gqlclient graphql.Client, q *queueValues, isOccurrence model.NeighborsNeighborsIsOccurrence) {
+	// TODO: take into account PkgEqual case where there is another alias of the artifact package that we don't know about
+	// Right now we only explore isOccurrences if their artifact is from a HasSLSA
+	if q.nowNode.slsaSubject {
+		switch subject := isOccurrence.Subject.(type) {
+		case *model.AllIsOccurrencesTreeSubjectPackage:
+			addNodeToQueue(q, q.now, q.nowNode.Depth, PackageVersion, false, []string{subject.Namespaces[0].Names[0].Versions[0].Version}, subject.Namespaces[0].Names[0].Versions[0].Id)
+			addNodeToQueue(q, q.now, q.nowNode.Depth, PackageName, false, []string{subject.Namespaces[0].Names[0].Versions[0].Version}, subject.Namespaces[0].Names[0].Id)
+		case *model.AllIsOccurrencesTreeSubjectSource:
+			addNodeToQueue(q, q.now, q.nowNode.Depth, SourceName, false, nil, subject.Namespaces[0].Names[0].Id)
 		}
-		q.nodeMap[isDependency.Package.Namespaces[0].Names[0].Id] = dfsNName
 	}
+}
 
-	if !dfsNVersion.Expanded {
-		q.queue = append(q.queue, isDependency.Package.Namespaces[0].Names[0].Versions[0].Id)
-	}
+func addNodeToQueue(q *queueValues, parent string, depth int, Type NodeType, slsa bool, versions []string, id string) {
```

Comment:
this could be a pointer receiver for  the queue struct, so you don't have to pass in all the variables.

Reply:
i dont believe this is resolved. Refer to https://go.dev/tour/methods/4

Reply:
I think this is an outdated version of the file, on the latest I have this: 
func addNodeToQueue(q *queueValues, nodeType NodeType, versions []string, id string)
so I think this follows the same pointer receiver for the struct as you linked instead of passing in all the variables?

Reply:
ok that's fine